spectral_cube/dask_spectral_cube.py

```diff
@@ -80,7 +80,7 @@ def wrapper(self, *args, **kwargs):
                 raise ImportError("saving the cube to a temporary directory "
                                   "requires the zarr and fsspec packages to "
                                   "be installed.")
-            filename = tempfile.mktemp()
+            filenum, filename = tempfile.mkstemp()
             with dask.config.set(**cube._scheduler_kwargs):
                 cube._data.to_zarr(filename)
             cube._data = da.from_zarr(filename)
```

Comment:
```suggestion
            cube._tmpdir = tempfile.TemporaryDirectory()
            with dask.config.set(**cube._scheduler_kwargs):
                cube._data.to_zarr(cube._tmpdir.name)
            cube._data = da.from_zarr(cube._tmpdir.namefile)
```
Slightly different option; this will enable the `TemporaryDirectory` to be cleaned up on deletion of the cube by setting up a custom teardown method for `DaskSpectralCubeMixin`:
```python
    def __del__(self):
        # Clean up any `tempfile.TemporaryDirectory` created by `save_to_tmp_dir`.
        if hasattr(self, '_tmpdir'):
            self._tmpdir.cleanup()
```